### Disclaimer : 
##### Data direcly taken from Marvel.com Character Index which includes "team" names and are not technically superhero names. Its hard to manually check and remove them all, so I am ignoring such inconsistencies

### Import Names 

In [107]:
import pandas as pd
# Read the CSV file into a pandas data frame (df)
heros = pd.read_csv('MarvelSuperHeros.csv', delimiter=',' ,header=None)

### Dataset Size

In [108]:
#number of super heros
heros.size

1491

### Lets have a look at few heros

In [109]:
heros.sample(10)

0
740      Magma (Amara Aquilla)
964                      Pride
204                    Cardiac
1029                   Redwing
945               Peter Parker
1122           She-Hulk (Lyra)
724                Machine Man
1218                   Stature
784   Maverick (Chris Bradley)
452                    Glorian

### oops! Gotta get rid of those extra info in brackets and strip trailing spaces that are left

In [110]:
heros=heros[0].str.split("(",expand=True)
#strip space XXX" ""(YYY)-->XXX and not XXX" "
heros=pd.DataFrame(heros[0])
heros=heros[0].str.strip()
heros=pd.DataFrame(heros)

In [111]:
heros.sample(10)

0
260                    Crule
196           Captain Marvel
450                Gladiator
693          Living Tribunal
943             Pet Avengers
163   Brotherhood of Mutants
1390          Venus Dee Milo
1090               Scrambler
897                Nightmare
759                Marauders

### Fixed? Not yet, Unfortunately we have more than 1 actors for each Hero. So, lets remove dups

In [112]:
print "before :" ,heros.size
heros = heros.drop_duplicates()
print "after :" , heros.size

before : 1491
after : 1257


### Should We have "-" or no? I am keep it ;)

In [113]:
heros

0
0                     3-D Man
1                      A-Bomb
2                      A.I.M.
3                 Aaron Stack
4                 Abomination
6               Absorbing Man
7                       Abyss
9                Adam Destine
10               Adam Warlock
11                      Aegis
12                Agent Brand
13                    Agent X
14                 Agent Zero
15            Agents of Atlas
16                     Aginar
17                 Air-Walker
18                       Ajak
19                     Ajaxis
20                      Akemi
21                      Alain
22              Albert Cleary
23                     Albion
24                 Alex Power
25                Alex Wilder
26               Alexa Mendez
27           Alexander Pierce
28                      Alice
29             Alicia Masters
30               Alpha Flight
32                Alvin Maker
...                       ...
1460                  Wrecker
1461            Wrecking Crew
1462                     X-23
1463                     X-51
1464                 X-Babies
1465               X-Cutioner
1466                 X-Factor
1467  X-Factor Investigations
1468                  X-Force
1469                    X-Man
1470                    X-Men
1472                    X-Ray
1473                 X-Statix
1474                   X.S.E.
1475                    Xavin
1476                     Xorn
1477              Yellow Claw
1478             Yellowjacket
1479           Young Avengers
1480              Young X-Men
1481                 Zaladane
1482                    Zaran
1483                    Zarda
1484                    Zarek
1485                 Zeigeist
1486                     Zemo
1487                   Zodiak
1488                   Zombie
1489                    Zuras
1490                    Zzzax

[1257 rows x 1 columns]

In [114]:
heros.to_csv("SuperHeroCorpus.csv",sep=',', encoding='utf-8', index=False, header=None)

### We now have a dataset of 1257 names.... It would be cool if we could visulaize the words and their 

### similarity, patterns? 

### Let's create our own Super Hero!